### PR TITLE
DAOS-12072 control: Check client cert dir on startup

### PR DESCRIPTION
**Pull request by kjacque**
- Verify the client certificate directory is accessible during DAOS server startup.
- Provide a more descriptive error if the client certs can't be loaded at runtime.

Features: security control

Required-githooks: true

Signed-off-by: Kris Jacque <kristin.jacque@intel.com>

### Before requesting gatekeeper:

* [x] Two review approvals and any prior change requests have been resolved.
* [x] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Comment by kjacque** [comment]
EL8 Functional security failures are a known issue: https://daosio.atlassian.net/browse/DAOS-12501

**Review by tanabarr — APPROVED** [review]

**Review by knard-intel — APPROVED** [review]
LGTM
Message errors improvement should be of great helps for enduser and sustaining tasks 😄

**Comment by kjacque** [comment]
The only test failure is a known issue in the control ftest suite: https://daosio.atlassian.net/browse/DAOS-12013

File: src/include/daos_errno.h [hunk 1]
```diff
@@ -285,7 +285,9 @@ extern "C" {
 	ACTION(DER_UPDATE_AGAIN,	(DER_ERR_DAOS_BASE + 41),	\
 	       update again)						\
 	ACTION(DER_NVME_IO,		(DER_ERR_DAOS_BASE + 42),	\
-	       NVMe I/O error)
+	       NVMe I/O error)						\
+	ACTION(DER_NO_CERT,		(DER_ERR_DAOS_BASE + 43),	\
+	       Unable to access one or more certificates)
```
Comment (mjmac) [inline src/include/daos_errno.h:290]: I was a little confused by this... This errno is eventually returned to the client application if the credentials are unable to be verified on the server side, right?

Comment (kjacque) [reply src/include/daos_errno.h:290]: No, this one is for the case where the cert can't be loaded on the agent side at all. I'm working on another PR for the invalid/incompatible cert case.

Comment (mjmac) [reply src/include/daos_errno.h:290]: Ah, ok. But the point is that the errno will be returned to the client application, right? i.e. We're not adding a new errno just for control plane tests?

Comment (kjacque) [reply src/include/daos_errno.h:290]: Yes, it is returned to the client application.

**Pull request merged**